File: algorithm_w/src/term.rs
pub enum Term {
    Var(String),                      // x
    Lit(String),                      // 1, true
    Abs(String, Box<Term>),           // λ x. t
    App(Box<Term>, Box<Term>),        // (t, t)
    Let(String, Box<Term>, Box<Term>) // let x = t1 in t2
}

File: Sources/AlgorithmW/Term.swift
//
// Created by ukitaka on 2017/10/12.
//

public indirect enum Term {
    case `var`(VarName)
    case literal(Literal)
    case abs(VarName, Term)
    case app(Term, Term)
    case `let`(VarName, Term, Term)
}
